### DISCUSSION TITLE
Examples on using ESP32-S3-BOX

### DISCUSSION THREAD

**Discussion by dac2020** [arendst/Tasmota#20855] [Q&A] [answered]
I've converted the ESP32-S3-BOX to tasmota and now need to learn more about how I can display images, text etc. 
There are some things I found on the Internet but not much. I also found the display page for tasmota but it doesn't help (yet) since I need some starting points first. 

I've also read something where someone was saying flashed with tasmota, it should now show a logo or something on boot but it's just a black screen so far. 

Does anyone know of an article or document that shares some examples on how I can use it? As with other devices I started using, once I have some basic understandings of using it, I can expand my knowledge from there.

Thanks for any help.

**Comment by s-hadinger** [comment]
Start here https://tasmota.github.io/docs/HASPmota/

**Comment by dac2020** [comment]
Hi thank you. 

I did as suggested at the start of the doc and it seemed to work but the screen remains black on boot. 
Seems there is a file missing or something missing in the document, a step I need to do.

```
00:00:00.001 CMD: Using USB CDC
00:00:00.001 HDW: ESP32-S3 v0.2 (PSRAM)
00:00:00.015 UFS: FlashFS mounted with 12588 kB free
00:00:00.023 CFG: Loaded from File, Count 15
00:00:00.030 QPC: Count 3
00:00:00.034 I2C: Bus1 using GPIO40(SCL) and GPIO41(SDA)
00:00:00.034 I2C: Bus2 using GPIO18(SCL) and GPIO08(SDA)
>>>>>00:00:00.038 TFS: File 'mcp23x.dat' not found
00:00:00.068 BRY: Berry initialized, RAM used 5996 bytes
00:00:00.960 DSP: Configured display 'ILI9342'
00:00:00.966 Project tasmota - Tasmota Version 13.4.0.2(6a35795-lvgl-haspmota)-2_0_14(2024-03-02T07:55:12)
00:00:00.971 LVG: LVGL initialized
00:00:02.000 WIF: Connecting to AP1 gst2 Channel 1 BSSId A4:2B:B0:FA:3C:AB in mode 11n as tasmota-5ABF70-8048...
00:00:03.828 WIF: Connected
16:28:50.008 HTP: Web server active on tasmota-5ABF70-8048 with IP address 192.168.254.100
16:28:51.074 RSL: INFO1 = {"Info1":{"Module":"ESP32S3-SBOX","Version":"13.4.0.2(6a35795-lvgl-haspmota)","FallbackTopic":"cmnd/DVES_5ABF70_fb/","GroupTopic":"cmnd/tasmotas/"}}
16:28:51.076 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-5ABF70-8048","IPAddress":"192.168.254.100","IP6Global":"","IP6Local":"fe80::3230:f9ff:fe5a:bf70%st1"}}
16:28:51.077 RSL: INFO3 = {"Info3":{"RestartReason":"Vbat power on reset","BootCount":8}}
16:28:51.078 RSL: RESULT = {"POWER":"ON"}
16:28:51.078 RSL: POWER = ON
16:28:53.935 QPC: Reset
16:28:55.931 RSL: STATE = {"Time":"2024-03-03T16:28:55","Uptime":"0T00:00:09","UptimeSec":9,"Heap":211,"SleepMode":"Dynamic","Sleep":10,"LoadAvg":99,"MqttCount":0,"Berry":{"HeapUsed":5,"Objects":58},"POWER":"ON","Wifi":{"AP":1,"SSId":"gst2","BSSId":"A4:2B:B0:FA:3C:AB","Channel":1,"Mode":"11n","RSSI":98,"Signal":-51,"LinkCount":1,"Downtime":"0T00:00:03"}}
16:28:55.966 RSL: SENSOR = {"Time":"2024-03-03T16:28:55","ANALOG":{"A1":3870}}
```
I seem to be using the correct firmware.

![image](https://github.com/arendst/Tasmota/assets/9727841/4454d6a1-da85-4d2e-bdb1-9ef2aceadaf2)

**Comment by dac2020** [comment]
Wait, I think I see what the problem is.
>Firmware: HASPmota is included in tasmota32-lvgl firmwares.

I'm using the wrong firmware perhaps. I used the default one in the tasmota installer.
Which ever I use must have TLS for MQTT otherwise, I won't be able to remotely control the display.

**Reply by sfromis** [reply]
I explicitly told you to not use the default selection, but a specific one. And what you posted is not using the default one. `13.4.0.2(6a35795-lvgl-haspmota)` should be ok.

**Reply by sfromis** [reply]
If the configuration is ok, you should *briefly* see a Tasmota logo, and/or the display name after boot.

Did you follow my suggestion about using auto-configuration? That was meant to be done *instead* of using the template on the templates repository page.

**Comment by dac2020** [comment]
This is what you told me in another post and is what I did.

```
What I'd recommend would be to config it for [Universal Display](https://tasmota.github.io/docs/Displays/#universal-display-driver) (already done below), and use [HASPmota](https://tasmota.github.io/docs/HASPmota/) or [LVGL](https://tasmota.github.io/docs/LVGL/), instead of the old-style display commands.

First you need a relevant binary, probably something like tasmota32s3cdc-qio_opi.factory.bin from
https://github.com/tasmota/install/tree/firmware/firmware/unofficial which you can select in the [Tasmota Web Installer](https://tasmota.github.io/install/) as "Tasmota32-S3 (QIO Flash/OPI PSRAM). That image includes support for LVGL and HASPmota, and I'd expect it to enable PSRAM for your device. Next, run Auto-configuration selecting "BOX". This installs a template including enabling Universal Display, and the needed display.ini file.
```

By default, I mean the Tasmota32-S3(Q10 Flash/OPI PSRAM), the only one in the web installer and then I used auto configure.
And no, I did not use the template found in the repo.

**Comment by s-hadinger** [comment]
You can safely ignore `00:00:00.038 TFS: File 'mcp23x.dat' not found`

**Comment by dac2020** [comment]
If the missing file is not a problem, then I'm missing something else as nothing has shown up on the screen at all. 
Do I have to power up the screen or something? All I've done is connect a good 5V supply to it and the WiFi comes up so I can console in but screen is always blank.

**Reply by sfromis** [reply]
Not sure what the backstory is, but while the autoconf setup is for a ILI9242 display, the template page suggests ST7789 instead. Both could make sense, so the question may be what is going on, like if  two versions have been created, or .....

Espressif does have pages seemingly confirming display driver IC to be ILI9342C like [this](https://github.com/espressif/esp-box/blob/master/docs/hardware_overview/esp32_s3_box/hardware_overview_for_box.md).

**Comment by dac2020** [comment]
So there is a bug, mismatch, something out of my control then? I just need to wait and see how this gets updated then try again?

**Reply by sfromis** [reply]
Well, sitting around and waiting is not much of a strategy. If there is a mismatch, "someone" better analyze what happens, which is "less likely" without having the hardware at hand.

My base assumption would be that both very different setups were created on the basis of having it working, but I don't know any details outside of finding the configs.

**Comment by dac2020** [comment]
I'm not a developer, only a supporter of open source which I try to excite everyone I can about so I don't think I can solve the problem, only find it.  Since I have the hardware, I'm happy to try some ideas, just not sure what however.

I did try the template from the template page but that didn't change anything.

**Comment by dac2020** [comment, marked as answer]
I gave up on this and was lucky enough to send it back. If I see updates about it, I might buy another one. 
Thanks for the help.
